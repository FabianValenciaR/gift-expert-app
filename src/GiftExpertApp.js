import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GiftGrid } from './components/GiftGrid';

export const GiftExpertApp = () => {
    // const categories = ['One Punch', 'Samurai x', 'Dragon Ball'];
    const [categories, setCategories] = useState(['Dragon Ball']);

    // const handleAdd = () =>{
    //     // setCategories([...categories, 'Hunter']);
    //     setCategories(cat => {
    //         return [...cat, 'Hunter'];
    //     });
    // }

    return (
        <>
            <h2>GiftExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />


            <ol>
                {
                    categories.map((category, i) => (
                        <GiftGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    )
}
