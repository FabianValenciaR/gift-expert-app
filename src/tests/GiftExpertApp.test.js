import { shallow } from "enzyme"
import { GiftExpertApp } from "../GiftExpertApp"
import React from 'react';

describe('Pruebas de <GiftExpertApp />', () => {
    test('Debe de renderizar el componente', () => {
        const wrapper = shallow(<GiftExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar una lista de categorias', () => {
        const categories = ['Dragon Ball', 'One Punch'];
        const wrapper = shallow(<GiftExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GiftGrid').length).toBe(categories.length);
    })


})
