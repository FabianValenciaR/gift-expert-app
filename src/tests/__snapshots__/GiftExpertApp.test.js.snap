// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GiftExpertApp /> Debe de renderizar el componente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas de <GiftExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiftGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GiftGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
